app/models/user.rb

```diff
@@ -167,6 +167,32 @@ def issue_membership_number
     self.membership_number = self.membership_number.blank? ? get_next_membership_number : self.membership_number
   end
 
+  def self.condition_response(condition, log)
+
+    case condition.name
+    when 'membership_will_expire'
+      case condition.timing
+      when 'before'
```

Comment:
'before' should be a constant.  (As should 'after')

these apply to all `Conditions` yes?  So that's where the constant should be defined.

Reply:
I went back and forth on this and decided against specifying constants.

1. The `timing` attribute is meant to an aid to the developer in writing clearer code.  For example, perhaps we have a response class that handles both the "before" and "after" processing for a specific condition (perhaps before and after membership expiration).  Using the `timing` attribute allows for clearer switching logic.  There is nothing special about the `timing` value - it is "free-form" and specific to the developer's response code. 

2. The `timing` idea could actually be embedded in the `config` attribute (and perhaps should be).  The only reason I pulled that out as a separate attribute is that most conditions responses will be related to time (before, after, every-day, the-day-of, last-quarter-of-year, etc.).

3. I also like that this makes the records easier to read, and convey intent, e.g.
```
class_name: 'MembershipExpireAlert',
  name: 'membership_will_expire',
  timing: 'before',
  config: { days: [60, 30, 14, 2] }

class_name: 'MembershipExpireAlert',
  name: 'membership_has_expired',
  timing: 'after',
  config: { days: [1, 7, 30] }

class_name: 'CompanyActivityAlert',
  name: 'company_visitor_activity',
  timing: 'every Friday',
  config: { activity_previous_days: 7, unique_visitor_threshold: 10 }

class_name: 'UpcomingEvent',
  name: 'company_upcoming_event',
  timing: 'before',
  config: { days: [7, 2] }
```
Of course, if we decide to limit each class to handling only _one_ condition, then we don't need `name` either.

That being said, I could go either way(s) on this.  Let's discuss at the meeting today.